core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -333,8 +333,9 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
       }
     try {
       val appListener = new ApplicationEventListener
+      val maybeTruncated = logPath.getName().endsWith(EventLoggingListener.IN_PROGRESS)
```

Comment:
This needs to call `isApplicationCompleted(eventLog)` instead, to handle legacy logs. (You can then replace the same call a few lines below with the result, to avoid another trip to HDFS.)